explainer.md

```diff
@@ -1,624 +1,411 @@
 
-### New Document
-
-```html
-<html>
-<head>
-  <style>
-    body {
-      background-color: lightblue;
-    }
-    .animated {
-      position: relative;
-      top: 100px;
-      width: 200px;
-      height: 200px;
-      background-color: green;
-    }
-
-    ::shared-container(header-id) {
-      animation: ::shared-container-header-id 1s ease-in;
-    }
-  </style>
-</head>
-<body>
-  <div class="animated" sharedid="header-id" id="header">Shared Element</div>
-</body>
-</html>
-```
+This section covers the concepts and mechanisms, while a later section looks at possible API shapes.
+
+Performing a transition from Page-A to Page-B requires parts of both to be on screen at the same time, potentially moving independently. This is currently impossible in a cross-document navigation, but it's still hard in an SPA (single page app) navigation. You need to make sure that the outgoing state persists along with the incoming state, that it can't receive additional interactions, and ensure the presence of both states doesn't create a confusing experience for those using accessibility technology.
+
+The aim of this design is to allow for representations of both Page-A and Page-B to exist at the same time, without the usability, accessibility, performance, security and memory concerns of having both complete DOM trees alive.
+
+Here's the example that will be used to explain the design:
+
+<img alt="Page-A and Page-B" src="media/pages.png?raw=true">
+
+The concepts and process described in this section apply for both MPA and SPA transition, however the API will differ in parts.
+
+Cross-origin transitions are something we want to tackle, but may have significant differences and restrictions for security reasons. This, cross-origin transitions are not covered in this document.
+
+## Part 1: The offering
+
+Before Page-A goes away, it offers up elements to be used in the transition. Generally, this will mean an element that animates independently during the transition. For the example transition, the elements are:
+
+- The header
+- The share button
+- The rest (referred to as the page root)
+
+https://user-images.githubusercontent.com/93594/141104275-6d1fb67a-2f73-41e4-9cef-14676798223b.mp4
+
+Aside from the root, an element offered for a transition has the following restrictions:
+
+- [`contain: paint`](https://developer.mozilla.org/en-US/docs/Web/CSS/contain) which ensures that the element is the containing block for all positioned descendants and generates a stacking context. This implies that the child content will be clipped to the context-box but it can be expanded using ['overflow-clip-margin'](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-clip-margin). Being a stacking context and a containing block allows the element to be treated as a single unit, whereas paint containment simplifies implementation.
```

Comment:
```suggestion
- [`contain: paint`](https://developer.mozilla.org/en-US/docs/Web/CSS/contain) which ensures that the element is the containing block for all positioned descendants and generates a stacking context. This implies that the child content will be clipped to the context-box but it can be expanded using ['overflow-clip-margin'](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-clip-margin). Being a stacking context and a containing block allows the element to be treated as a single unit which simplifies implementation.
```